pkg/sql/ordering.go

```diff
@@ -24,35 +24,83 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 )
 
-// orderingInfo describes the column ordering on a set of results.
+// orderingInfo describes the column ordering on a set of results. Typically the
+// ordering information is used to see to what extent it can satisfy a "desired"
+// ordering (which is just a list of columns and directions).
 //
-// If results are known to be restricted to a single value on some columns, we call these "exact
-// match" columns; these are inconsequential w.r.t ordering.
+// In its simplest form, an ordering is simply a list of columns and a direction
+// for each column, for example a+,b-,c+. This indicates that the rows are
+// ordered by the value on column a (ascending); rows that have the same value
+// on a are ordered by column b (descending); and rows that have the same values
+// on a and b are ordered by column c (ascending).
 //
-// For example, if an index was defined on columns (a, b, c, d) and the WHERE clause was
-// "(a, c) = (1, 2)" then a and c are exact-match columns and we have an ordering by b then d.
-// Such an ordering satisfies any of the following desired orderings (among many others):
-//    a, c
-//    c, a
-//    b, a, c
-//    b, c, a
-//    a, b, c
-//    c, b, a
-//    b, d, a
-//    a, b, c, d
-//    b, a, c, d
+// == Exact match columns ==
```

Comment:
I think this PR may also be the opportunity to rename this concept from "exact match" (which is confusing) to "known-constant" (which actually reflects the property).

For example `SELECT 42 AS x, k FROM kv` will have x known-constant, although there is no "match" to speak of.